File: yaw_phonenumber/android/src/main/kotlin/app/illabo/yaw_phonenumber/YawPhonenumber.kt
package app.illabo.yaw_phonenumber

import android.content.Context
import android.os.Build
import io.michaelrocks.libphonenumber.android.PhoneNumberUtil
import java.util.*

class YawPhonenumber(private val ctx: Context) {
    val phoneUtil = PhoneNumberUtil.createInstance(ctx)

    fun formatIncomplete(string: String): String {
        val locale = getCurrentLocale(ctx)
        val ayt = phoneUtil.getAsYouTypeFormatter(locale.country)
        var formated =""
        string.forEach {
            formated = ayt.inputDigit(it)
        }

        return formated
    }

    fun formatAsInternational(string: String): String{
        val locale = getCurrentLocale(ctx)
        val finalPhone = try {
            phoneUtil.parse(string, locale.country)
        } catch (t: Throwable) {
            return string
        }
        return phoneUtil.format(
                finalPhone,
                PhoneNumberUtil
                .PhoneNumberFormat
                .INTERNATIONAL)
    }

    fun isValidNumber(string: String): Boolean{
        val locale = getCurrentLocale(ctx)
        val finalPhone = try {
            phoneUtil.parse(string, locale.country)
        } catch (t: Throwable) {
            return false
        }
        return phoneUtil.isValidNumber(finalPhone)
    }

    fun isInternationallyDialable(string: String): Boolean{
        val locale = getCurrentLocale(ctx)
        val finalPhone = try {
            phoneUtil.parse(string, locale.country)
        } catch (t: Throwable) {
            return false
        }
        return phoneUtil.canBeInternationallyDialled(finalPhone)
    }

    private fun getCurrentLocale(context: Context): Locale {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            return context.resources.configuration.locales.get(0)
        } else {
            @Suppress("DEPRECATION")
            return context.resources.configuration.locale
        }
    }
}

File: yaw_phonenumber/lib/yaw_phonenumber.dart
import 'dart:async';

import 'package:yaw_phonenumber_platform_interface/yaw_phonenumber_platform_interface.dart';

final YawPhonenumberPlatform instance = YawPhonenumberPlatform.instance;

Future<String> formatIncomplete(String string) async {
  return await instance.formatIncomplete(string);
}

Future<String> formatAsInternational(String string) async {
  return await instance.formatAsInternational(string);
}

Future<bool> isValidNumber(String string) async {
  return await instance.isValidNumber(string);
}

Future<bool> isInternationallyDialable(String string) async {
  return await instance.isInternationallyDialable(string);
}
